src/main/java/com/google/collegeplanner/servlets/SchedulerServlet.java

```diff
@@ -0,0 +1,140 @@
+      selectedClasses = (JSONArray) body.get("selectedClasses");
+    } catch (NumberFormatException | ClassCastException | ParseException | NullPointerException e) {
+      respondWithError(
+          "Invalid body for POST request.", HttpServletResponse.SC_BAD_REQUEST, response);
+      return;
+    }
+    
+    try {
+      prepareLists(selectedClasses, response);
+    } catch (IOException e ) {
+      return;
+    }
+
+    JSONObject schedules = getSchedules();
+    response.setContentType("application/json;");
+    response.getWriter().println(new Gson().toJson(schedules));
+  }
+
+  /**
+   * This method loads the courses and courseList ArrayLists with correct
+   * information. courses is loaded with the lists of sections for each
+   * course while courseList is loaded with the courseID's of the selected
+   * courses.
+   * @param classes The JSONArray of courseIds of the selected classes
+   * @param response The HttpServletResponse object
+   */
+  private void prepareLists(JSONArray classes, HttpServletResponse response) throws IOException {
+    String courseId;
+    URI uri;
+    JSONArray jsonArray;
+    ApiUtil apiUtil = new ApiUtil();
+
+    courseList = new ArrayList<String>();
+    courses = new ArrayList<ArrayList<Section>>();
+    for (Object obj : classes) {
+      courseId = (String) obj;
+      courseList.add(courseId);
+      try {
+        uri = new URI("https://api.umd.io/v1/courses/" + courseId + "/sections");
+      } catch (URISyntaxException e) {
+        respondWithError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, response);
+        throw new IOException(courseId + " cannot be found in the umd.io database");
+      }
+
+      jsonArray = apiUtil.getJsonArray(uri);
+      if (jsonArray == null) {
+        respondWithError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, response);
+        return;
+      }
+      courses.add(convertSectionJsonToSectionArrayList(jsonArray));
+    }
+  }
+
+  /**
+   * Returns a JSONArray representing the schedules the
+   * SemesterScheduler returns.
+   */
+  private JSONObject getSchedules() {
+    JSONObject json = new JSONObject();
+    JSONArray schedulesJson = new JSONArray();
+    SemesterScheduler scheduler = new SemesterScheduler(courses);
+    ArrayList<Schedule> possibleSchedules = scheduler.getPossibleSchedules();
+
+    for (Schedule schedule : possibleSchedules) {
+      schedulesJson.add(schedule.toJSON());
+    }
+
+    json.put("schedules", schedulesJson);
+
+    return json;
+  }
+
+  /**
+   * Converts a given JSONArray of sections into an ArrayList of
+   * sections.
+   * @param json The JSONArray of Sections that will be converted into an
+   * ArrayList of Sections
+   */
+  private ArrayList<Section> convertSectionJsonToSectionArrayList(JSONArray json) {
+    ArrayList<Section> list = new ArrayList<Section>();
+    for (Object obj : json) {
+      try {
+        list.add(new Section((JSONObject) obj));
+      } catch (Exception e) {
```

Comment:
We should catch a narrower exception. Please see the following excerpt from the Java readability guidelines:

```
Does the underlying code actually throw Exception, or a derived type of exception? You should generally catch the narrowest exception type.

Only catch Exception when you legitimately want to handle any exception that may arise, including unchecked ones you don't expect to actually happen.

Unchecked exceptions are usually used to convey the message that either the exception can be prevented by proper use of the API, or that callers should rarely catch this exception because there's nothing they can do about it. However, `catch (Exception e)` is fine if your intention is to gracefully downgrade the flow regardless (e.g., a catch-all in the main() method, which prevents any exceptions from being propagated to the Java VM).
```

Reply:
I really can't make it any narrower as the exception being thrown is a java.text.ParseException. However, I already have org.json.simple.parser.ParseException being imported.

Reply:
To solve the duplicate name, you can import `java.text`, and reference `text.ParseException` -- or vice versa.